src/app/api/analytics/background_task.py

```diff
@@ -129,10 +138,88 @@ def run():
         }
 
         oer_search_store.append(
-            SearchStore(node_id=row.id, missing_materials=missing_materials)
+            SearchStore(node_id=collection.id, missing_materials=missing_materials)
         )
 
+        materials = [
+            PendingMaterials(**build_pending_materials(collection))
+            for collection in sub_collections
+        ]
+
+        pending_materials_store.append(
+            PendingMaterialsStore(
+                collection_id=collection.id, missing_materials=materials
+            )
+        )
+
+    logger.info("Storing in global store.")
+
     global_store.search = search_store
```

Comment:
I am surprised this works as the global_store dataclass is declared frozen o.0

Reply:
nevermind, i just saw the frozen = False there and totally read it as Frozen=True

Reply:
Removed.